engine/runtime/src/main/java/org/enso/interpreter/node/expression/literal/IntegerLiteralNode.java

```
@@ -33,4 +34,13 @@ public static IntegerLiteralNode build(long value) {
   public Object executeGeneric(VirtualFrame frame) {
     return this.value;
   }
+
+  @Override
+  public Object parsePatch(String text) {
+    try {
+      return Long.valueOf(text);
+    } catch (NumberFormatException ex) {
```

Probably shouldn't silently ignore the exception. warning perhaps?

---

Right. We should use proper Enso parsing of a number just like in the [text literal case](https://github.com/enso-org/enso/pull/3508/files#r894227352). I've just noticed there is `Compiler.runInline` which calls 
```
val parsed: AST = parse(source)
generateIRInline(parsed).flatMap { ir =>
      val compilerOutput = runCompilerPhasesInline(ir, newContext)
      runErrorHandlingInline(compilerOutput, source, newContext)
      Some(truffleCodegenInline(compilerOutput, source, newContext))
}
```
I guess the `Patchable` concept shall do something similar.

---

We can reuse the `AstToIr.translateInline` method to parse these kinds of literals:
```scala
scala> org.enso.compiler.codegen.AstToIr.translateInline(new org.enso.syntax.text.Parser().run("2_1"))
val res4: Option[org.enso.compiler.core.IR.Expression] = Some(IR.Literal.Number(base = Some(2), value = 1, location = Some(IdentifiedLocation(Location(0,3),Some(a98a4dc4-3c12-425a-ac5c-2189bca58f3c))), passData = Iterable(), diagnostics = DiagnosticStorage(diagnostics = List()), id = 8f1c55e8-92fb-41ff-af8f-8f6ac51c03f6))

scala> org.enso.compiler.codegen.AstToIr.translateInline(new org.enso.syntax.text.Parser().run("\"Hello\""))
val res5: Option[org.enso.compiler.core.IR.Expression] = Some(IR.Literal.String(text = Hello, location = Some(IdentifiedLocation(Location(0,7),Some(bcbd238a-852c-490e-885b-a69a02977680))), passData = Iterable(), diagnostics = DiagnosticStorage(diagnostics = List()), id = bce7f279-a018-4542-af2b-7539dc0ca36a))
```

---

Done in [a84ae04](https://github.com/enso-org/enso/pull/3508/commits/a84ae0425070de185501ea02fc54cbc9e0ab7188) - thanks for the investigation.